pip_audit/_service/pypi.py

```diff
@@ -109,12 +109,11 @@ def _get_pip_cache() -> str:
 
 
 def _get_cache_dir(custom_cache_dir: Optional[Path]) -> str:
-    pip_cache_dir: Optional[str] = (
-        _get_pip_cache() if _PIP_VERSION >= _MINIMUM_PIP_VERSION else None
-    )
     if custom_cache_dir is not None:
         return str(custom_cache_dir)
-    elif pip_cache_dir is not None:  # pragma: no cover
+
+    pip_cache_dir = _get_pip_cache() if _PIP_VERSION >= _MINIMUM_PIP_VERSION else None
```

Comment:
Typing should be detected here to be `Optional[str]`, but I'm not sure if the convention is to explicitly write it as: 

```python
pip_cache_dir: Optional[str] = _get_pip_cache() if _PIP_VERSION >= _MINIMUM_PIP_VERSION else None
```

Let me know and I'll change it!

Reply:
Looks good as-is 🙂  -- we don't have a formal style preference for explicit/implicit types, but the general pattern so far has only been to use explicit types when `mypy` needs them.